### PR TITLE
Disable vertical scroll only if scrolling horizontally

### LINKED ISSUE #4: preventEventOnTouchMove prevent vertical scroll of the page
Hi,

I have a list of carousels in mobile view.
When page scroll starts at the carousel it got prevented, there is any way to allow it?

### PR DESCRIPTION
**Pull request by develohpanda**
Fixes #4 

Preventing the event on `touchMove` should only happen when the swipe direction is horizontal, and it should not prevent if a vertical scroll is initiated on the carousel.

The current behavior prevents a vertical page scroll if the scroll is initiated from a carousel.

Happy to discuss this solution, however it's definitely a blocker for me using this across all of our projects. `react-alice-carousel` is really great!

😄

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ type ConstructorProps = {
   delta?: number | 10; // minimum distance to the start of the swiping (px)
   mouseTrackingEnabled?: boolean | false;
   touchTrackingEnabled?: boolean | true;
-  preventDefaultTouchmoveEvent?: boolean | false; // stop the browser scrolling while swiping
+  preventDefaultTouchmoveEvent?: boolean | false; // stop the browser scrolling vertically while swiping horizontally
```
Comment (maxmarinich) [inline README.md:18]: the option should prevent scrolling while any swiping

File: src/index.ts [hunk 2]
```diff
@@ -126,7 +126,9 @@ export default class VanillaSwipe {
     const { absX, absY, deltaX, deltaY, duration } = this.getPosition(e);
     const { delta, onSwiping, preventDefaultTouchmoveEvent } = this.props;
 
-    if (e.cancelable && preventDefaultTouchmoveEvent) e.preventDefault();
+    if (e.cancelable && preventDefaultTouchmoveEvent && !Utils.isVerticalTouchMoveDetected(deltaX, deltaY)) {
+      e.preventDefault();
```
Comment (maxmarinich) [inline src/index.ts:130]: What happen if user use the lib for vertical swiping?

**Review by maxmarinich — COMMENTED** [review]
Explanations are needed.

**Comment by maxmarinich** [comment]
closed as irrelevant

**Comment by develohpanda** [comment]
> closed as irrelevant

Sorry about that, got bogged down with other things. I ended up going ahead with https://kenwheeler.github.io/slick/ as it contains the expected behavior.

**Comment by felixmosh** [comment]
@maxmarinich can we split the `preventDefaultTouchmoveEvent` into 2 separate props?
One for preventDefaultTouch**Vertical**MoveEvent and one for preventDefaultTouch**Vertical**MoveEvent.
This way, in alice-react-carousel, on mobile view, when the carousel is horizontal, it will prevent only the preventDefaultTouch**Vertical**MoveEvent

WDYT?

**Pull request closed without merging**